### DISCUSSION TITLE
Question - Climate Control for Stream Deck Plus

### DISCUSSION THREAD

**Discussion by kramttocs** [cgiesche/streamdeck-homeassistant#202] [Help and Support]
Very excited to see Stream Deck Plus support.

Has anyone worked on setting up controlling a climate entity via a dial?
Not sure how to get started with making the dial execute the set target temperature service with [Current Temp] + or - [ticks]

Is it also possible to change the background dynamically based on the hvac mode? Flame, snowflake, etc

**Comment by jer78** [comment]
I posted something today that may help. It's using the dial to +/- for volume on a media_player. You could modify it for use with a climate device. The tricky part would be to limit it so it doesn't turn the temp to any extreme. Perhaps, have the automation first update a sensor and a few seconds delay before it actually sends to your thermostat. This would allow you to see what temperature it will change to before it sends that to your thermostat. If you do this, make sure the automation mode is set to "restart". 

https://github.com/cgiesche/streamdeck-homeassistant/issues/188#issuecomment-1836712121

**Comment by kramttocs** [comment]
Thank you! I'll look over that. Good call on the runaway part.  I wonder if logic could be added in the stream deck template where if the new value is > 75 (for example) it could just send 75. So keep the limit localized. Fortunately as a backup, most thermostats have a max and min threshold.

**Comment by jer78** [comment]
The logic pretty much has to be handled at the HA side, but that's ok because it's much easier to write templates there and there's a huge community always ready to help. With that said, let me know if you're stuck and I'll be happy to assist with the automation.

**Comment by kramttocs** [comment]
Yeah, I agree it'd be easier handling this in HA but I am not ready to write off doing it in the button logic yet :) Looking at the nunjucks referenced doc it allows for a decent amount of logic.
But...
I need to wrap my head around some of the more basic level templating/features of this plugin first and I appreciate the offer.

**Comment by cgiesche** [comment]
Hi kramttocs,

it is not possible to change the background image of the display via templating (in the plugin). But the plugin tries to use the icon that is provided by home assitant in the state attribute "icon". I am not a home-assistant expert, but maybe this may be a way to get state-dependent icons.

**Comment by Elitehoernchen** [comment]
> Very excited to see Stream Deck Plus support.
> 
> Has anyone worked on setting up controlling a climate entity via a dial? Not sure how to get started with making the dial execute the set target temperature service with [Current Temp] + or - [ticks]
> 
> Is it also possible to change the background dynamically based on the hvac mode? Flame, snowflake, etc

I am using a script for my smart thermostat entity to adjust the custom set temperature inside HA:

`alias: Temp
sequence:
  - service: climate.set_temperature
    data:
      temperature: >-
        {{ state_attr('climate.smart_thermostat_0300c267_none_0300c267',
        'temperature') + option }}
    target:
      entity_id: climate.smart_thermostat_0300c267_none_0300c267
  - service: climate.set_preset_mode
    data:
      preset_mode: custom
    target:
      entity_id: climate.smart_thermostat_0300c267_none_0300c267
mode: single
icon: mdi:thermometer-chevron-up
`

for configuration of the dial in StreamDeck configure the script for Rotation and add `{
"option" :  {{ticks}}
}` as Service Data JSON

**Reply by Elitehoernchen** [reply]
note: with changing the custom temperature i set the HVAC to custom. I got another script to cycle trough the HVAC mode ith dial press:

`alias: HeizungModus
sequence:
  - if:
      - condition: state
        entity_id: climate.smart_thermostat_0300c267_none_0300c267
        attribute: preset_mode
        state: custom
        for:
          hours: 0
          minutes: 0
          seconds: 5
    then:
      - service: climate.set_preset_mode
        data:
          preset_mode: auto
        target:
          entity_id: climate.smart_thermostat_0300c267_none_0300c267
      - stop: ""
    else: []
  - if:
      - condition: state
        entity_id: climate.smart_thermostat_0300c267_none_0300c267
        attribute: preset_mode
        state: auto
        for:
          hours: 0
          minutes: 0
          seconds: 5
    then:
      - service: climate.set_preset_mode
        data:
          preset_mode: comfort
        target:
          entity_id: climate.smart_thermostat_0300c267_none_0300c267
      - stop: ""
    else: []
  - if:
      - condition: state
        entity_id: climate.smart_thermostat_0300c267_none_0300c267
        attribute: preset_mode
        state: comfort
        for:
          hours: 0
          minutes: 0
          seconds: 5
    then:
      - service: climate.set_preset_mode
        data:
          preset_mode: sleep
        target:
          entity_id: climate.smart_thermostat_0300c267_none_0300c267
      - stop: ""
    else: []
  - if:
      - condition: state
        entity_id: climate.smart_thermostat_0300c267_none_0300c267
        attribute: preset_mode
        state: sleep
        for:
          hours: 0
          minutes: 0
          seconds: 5
    then:
      - service: climate.set_preset_mode
        data:
          preset_mode: away
        target:
          entity_id: climate.smart_thermostat_0300c267_none_0300c267
      - stop: ""
    else: []
  - if:
      - condition: state
        entity_id: climate.smart_thermostat_0300c267_none_0300c267
        attribute: preset_mode
        state: away
        for:
          hours: 0
          minutes: 0
          seconds: 5
    then:
      - service: climate.set_preset_mode
        data:
          preset_mode: auto
        target:
          entity_id: climate.smart_thermostat_0300c267_none_0300c267
      - stop: ""
    else: []
mode: single
icon: mdi:home-thermometer
`
